fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/jbatch/bulkdata/export/fast/ExportPartitionAnalyzer.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * (C) Copyright IBM Corp. 2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.jbatch.bulkdata.export.fast;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Logger;
+
+import javax.batch.api.partition.PartitionAnalyzer;
+import javax.batch.runtime.BatchStatus;
+import javax.batch.runtime.context.JobContext;
+import javax.enterprise.context.Dependent;
+import javax.inject.Inject;
+
+@Dependent
+public class ExportPartitionAnalyzer implements PartitionAnalyzer {
+    private static final Logger logger = Logger.getLogger(ExportPartitionAnalyzer.class.getName());
+
+    @Inject
+    JobContext jobContext;
+
+    // accumulate the individual PartitionSummary objects
+    private List<PartitionSummary> partitionSummaries = new ArrayList<>();
+
+    public ExportPartitionAnalyzer() {
+        // do nothing.
+    }
+
+    @Override
+    public void analyzeStatus(BatchStatus batchStatus, String exitStatus) {
+        // do nothing.
+    }
+
+    @Override
+    public void analyzeCollectorData(Serializable data) {
+        // data is the value returned by the ExportPartitionCollector
+        if (data == null) {
+            return;
+        }
+
+        PartitionSummary ps = (PartitionSummary) data;
+        logger.fine("adding partition summary to final result for '" + ps.getResourceType() + "'");
```

Comment:
should this get wrapped in a isLoggable, or is this only called once per job?

Reply:
only called once per resource type at the end of the job, so I skipped the barrier to keep the code cleaner.